cluster/manifests/skipper/deployment.yaml

```diff
@@ -1,35 +1,61 @@
 {{ $internal_version := "v0.18.42-691" }}
 {{ $version := index (split $internal_version "-") 0 }}
 
+{{ template "skipper-ingress" dict
+  "name" "skipper-ingress"
+  "internal_version" $internal_version
+
+  "Values" .Values
+  "ConfigItems" .ConfigItems
+}}
+
+{{ if .ConfigItems.skipper_ingress_canary_internal_version }}
+{{ template "skipper-ingress" dict
+  "name" "skipper-ingress-canary"
+  "internal_version" .ConfigItems.skipper_ingress_canary_internal_version
+  "replicas" 1
+  "args" (index .ConfigItems "skipper_ingress_canary_args")
+
+  "Values" .Values
+  "ConfigItems" .ConfigItems
+}}
+{{ end }}
+
+{{ define "skipper-ingress" }}
+{{ $version := index (split .internal_version "-") 0 }}
+---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: skipper-ingress
+  name: "{{ .name }}"
   namespace: kube-system
   labels:
     application: skipper-ingress
     version: {{ $version }}
     component: ingress
 spec:
+{{ if index . "replicas" }}
+  replicas: {{ .replicas }}
+{{ end }}
   strategy:
     rollingUpdate:
       maxSurge: 1
       maxUnavailable: 0
   selector:
     matchLabels:
-      deployment: skipper-ingress
+      deployment: "{{ .name }}"
   template:
     metadata:
       labels:
-        deployment: skipper-ingress
+        deployment: "{{ .name }}"
         application: skipper-ingress
         version: {{ $version }}
         component: ingress
       annotations:
         kubernetes-log-watcher/scalyr-parser: |
           [{"container": "skipper-ingress", "parser": "skipper-access-log"}]
         config/hash: {{"secret.yaml" | manifestHash}}
-        logging/destination: "{{.Cluster.ConfigItems.log_destination_local}}"
+        logging/destination: "{{ .ConfigItems.log_destination_local }}"
```

Comment:
> This is also not relevant change so please clean it up

It is because .ConfigItems is a template parameter not the global variable, that is why it is passed during template invocation as `"ConfigItems" .ConfigItems`

Reply:
true that makes sense. Then I would suggest to use something else as name like `CfgItems`, to be explicit that we are scoped.
wdyt?